lib/src/component_declaration/component_type_checking.dart

```diff
@@ -231,20 +232,27 @@ bool isComponentOfType(ReactElement instance, dynamic typeAlias, {
     return false;
   }
 
+  var finalInstance = instance;
+  var instanceType = finalInstance.type;
+
   var type = getComponentTypeFromAlias(typeAlias);
   if (type == null) {
     return false;
   }
 
-  var instanceType = instance.type;
+  if (instance.type != null && getProperty(instance.type, 'WrappedComponent') != null) {
+    finalInstance = getProperty(instance.type, 'WrappedComponent');
+    instanceType = instance.type.type;
```

Comment:
It seems like the goal with `instanceType` is always to keep it equal to `finalInstance.type`, but because `finalInstance` can change `instanceType` needs to be updated as well.

Is there a reason we can't just set `instanceType` once below this if block?